### ISSUE TITLE
bug(COMPONENT): TITLE

### ISSUE THREAD

**Issue by AugustZellmer** [angular/components#23244]
https://stackblitz.com/edit/components-issue-nel4pn

Steps to reproduce:
1. Create a mat-form-field with a mat-label and an input. Set the appearance type to "outline". Let the label value to a relatively long string.
2. Change the font to something with wide glyphs, like Poppins. Make sure it's imported from a 3rd party server that isn't *too* fast; something like Google Fonts.
3. Refresh the page.

 
 
#### Expected Behavior

The gap in the top border should we wide enough to accommodate the wide glyphs.
![1](https://user-images.githubusercontent.com/1058866/127019684-12cf313e-06f3-43ae-bbc8-44f258c2dd35.jpg)

#### Actual Behavior

The gap in the top border is not wide enough to accommodate the wide glyphs. It seems to calculate the width of the gap based on the browser's default font, before Poppins is fully downloaded. If Poppins is already in your browser's cache, and doesn't need to be redownloaded, everything works fine.
![2](https://user-images.githubusercontent.com/1058866/127020069-3af38f9a-fca3-4900-9a87-31b3b2e32529.jpg)

#### Environment

- Angular: 10.1.6 and 12.0.3 and others
- CDK/Material: 10.2.7 and 12.0.4 and others
- Browsers: Problem manifests on both Chrome and FF, but to different degrees due to the different default fonts on each browser. Other browsers untested.
- Operating System: Ubuntu. Other OSs untested.

**Comment by AugustZellmer** [comment]
Edit: This is def some sort of timing issue. The "break" in the top border is being calculated before the font by which it needs to be calculated has been downloaded.

**Comment by crisbeto** [comment]
Yeah, we try to measure as infrequently as possible in order to avoid triggering reflows. I believe to logic at the moment is to measure once on init and again if the text changes.

**Comment by AugustZellmer** [comment]
I didn't know about the `updateOutlineGap` function. That'll be helpful.

Whether this is a "bug" or an "optimization" is really up to you to decide. The issue only manifests if the custom font is not already in the browser's cache - ie. only the first time you load the page. So, do with this information what you see fit. Thanks!

**Comment by poveu** [comment]
@crisbeto is there any way to force `updateOutlineGap` on all fields without manually triggering it one by one, in every component that uses any kind of `MatFormField`?

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
